lightning/src/ln/msgs.rs

```diff
@@ -108,46 +181,36 @@ impl LocalFeatures {
 	}
 }
 
-/// Tracks globalfeatures which are in init messages and routing announcements
-#[derive(Clone, PartialEq, Debug)]
-pub struct GlobalFeatures {
-	#[cfg(not(test))]
-	flags: Vec<u8>,
-	// Used to test encoding of diverse msgs
-	#[cfg(test)]
-	pub flags: Vec<u8>
-}
-
-impl GlobalFeatures {
-	pub(crate) fn new() -> GlobalFeatures {
-		GlobalFeatures {
-			flags: Vec::new(),
-		}
+impl<T: FeatureContextInitNode> Features<T> {
+	pub(crate) fn supports_data_loss_protect(&self) -> bool {
+		self.flags.len() > 0 && (self.flags[0] & 3) != 0
 	}
 
-	pub(crate) fn requires_unknown_bits(&self) -> bool {
-		for &byte in self.flags.iter() {
-			if (byte & 0x55) != 0 {
-				return true;
-			}
-		}
-		return false;
+	pub(crate) fn supports_upfront_shutdown_script(&self) -> bool {
+		self.flags.len() > 0 && (self.flags[0] & (3 << 4)) != 0
 	}
+	#[cfg(test)]
+	pub(crate) fn unset_upfront_shutdown_script(&mut self) {
+		self.flags[0] ^= 1 << 5;
+	}
+}
 
-	pub(crate) fn supports_unknown_bits(&self) -> bool {
-		for &byte in self.flags.iter() {
-			if byte != 0 {
-				return true;
-			}
+impl Features<FeatureContextInit> {
+	pub(crate) fn initial_routing_sync(&self) -> bool {
+		self.flags.len() > 0 && (self.flags[0] & (1 << 3)) != 0
+	}
+	pub(crate) fn set_initial_routing_sync(&mut self) {
+		if self.flags.len() == 0 {
+			self.flags.resize(1, 1 << 3);
+		} else {
+			self.flags[0] |= 1 << 3;
 		}
-		return false;
 	}
 }
 
 /// An init message to be sent or received from a peer
 pub struct Init {
-	pub(crate) global_features: GlobalFeatures,
-	pub(crate) local_features: LocalFeatures,
+	pub(crate) features: Features<FeatureContextInit>,
```

Comment:
Consider making a type alias `InitFeatures` to remove duplication and verbosity.